src/storage/src/protocol/client.rs

```diff
@@ -273,19 +273,25 @@ impl Arbitrary for StorageCommand<mz_repr::Timestamp> {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct StorageFrontierUppers<T = mz_repr::Timestamp> {
+    pub data: Vec<(GlobalId, ChangeBatch<T>)>,
+    pub remap: Vec<(GlobalId, ChangeBatch<T>)>,
```

Comment:
Do we like this explicit listing of shard uppers. Especially if/when @brennanvincent has to add more shards when he adds support for plain `DEBEZIUM` envelope?